File: Conceitos de LPs/Subprogramas/Corrotinas/corrotinas.lua
//Lua

function tarefa1()
    for i = 1, 3 do
        print("Tarefa 1 - execução " .. i)
        coroutine.yield(os.execute("sleep " .. 1))
    end
end

function tarefa2()
    for i = 1, 2 do
        print("Tarefa 2 - execução " .. i)
        coroutine.yield(os.execute("sleep " .. 2))
    end
end

co1 = coroutine.create(tarefa1)
co2 = coroutine.create(tarefa2)

coroutine.resume(co1)
coroutine.resume(co2)
coroutine.resume(co1)
coroutine.resume(co2)
coroutine.resume(co1)

File: Conceitos de LPs/Subprogramas/Corrotinas/corrotinas.go
//GoLang

package main

import (
    "fmt"
    "time"
)

func tarefa1() {
    for i := 1; i <= 3; i++ {
        fmt.Printf("Tarefa 1 - execução %d\n", i)
        time.Sleep(1 * time.Second)
    }
}

func tarefa2() {
    for i := 1; i <= 2; i++ {
        fmt.Printf("Tarefa 2 - execução %d\n", i)
        time.Sleep(2 * time.Second)
    }
}

func main() {
    go tarefa1()
    go tarefa2()

    time.Sleep(5 * time.Second) // Aguarda a conclusão das corrotinas
}
